inc/datahandlers/warnings.php

```diff
@@ -339,6 +339,7 @@ function expire_warnings()
 			}
 		}
 
+		$plugins->run_hooks("datahandler_warnings_expire_warnings", $this);
```

Comment:
I correct myself; this should go before the query above because the while loop already modified the rows at this point.

Here, it is impossible to query for the warning identifiers to be affected.

Reply:
Maybe another solution would be to catch and pass on the modified warning identifiers, but I think just moving the hook is enough.